client/my-sites/checkout/plan-upgrade-nudge/index.jsx

```diff
@@ -0,0 +1,416 @@
+							src="/calypso/images/illustrations/themes.svg"
+						/>
+					</div>
+				</div>
+			</Fragment>
+		);
+	}
+
+	footer() {
+		const { translate } = this.props;
+		return (
+			<footer className="plan-upgrade-nudge__footer">
+				<Button
+					className="plan-upgrade-nudge__decline-offer-button"
+					onClick={ this.handleClickDecline }
+				>
+					{ translate( 'No thanks, I’ll stick with the free themes' ) }
+				</Button>
+				<Button
+					primary
+					className="plan-upgrade-nudge__accept-offer-button"
+					onClick={ this.handleClickAccept }
+				>
+					{ translate( 'Yes, give me those Premium designs!' ) }
+				</Button>
+			</footer>
+		);
+	}
+
+	render() {
+		const {
+			selectedSiteId,
+			isLoading,
+			hasProductsList,
+			hasSitePlans,
+			translate,
+			receiptId,
+		} = this.props;
+		const title = translate( 'Checkout ‹ Plan Upgrade', {
+			comment: '"Checkout" is the part of the site where a user is preparing to make a purchase.',
+		} );
+
+		return (
+			<Main className="plan-upgrade-nudge">
+				<PageViewTracker path="/checkout/:site/plan-upgrade-nudge/:receipt_id" title={ title } />
+				<DocumentHead title={ title } />
+				<QuerySites siteId={ selectedSiteId } />
+				{ ! hasProductsList && <QueryProductsList /> }
+				{ ! hasSitePlans && <QuerySitePlans siteId={ selectedSiteId } /> }
+
+				{ isLoading ? (
+					this.renderPlaceholders()
+				) : (
+					<>
+						{ receiptId ? (
+							<CompactCard className="plan-upgrade-nudge__card-header">
+								{ this.header() }
+							</CompactCard>
+						) : (
+							''
+						) }
+						<CompactCard className="plan-upgrade-nudge__card-body">{ this.body() }</CompactCard>
+						<CompactCard className="plan-upgrade-nudge__card-footer">{ this.footer() }</CompactCard>
+					</>
+				) }
+			</Main>
+		);
+	}
+}
+
+const trackUpsellButtonClick = buttonAction => {
+	// Track calypso_upgrade_plan_upsell_decline_button_click and calypso_upgrade_plan_upsell_accept_button_click events
+	return recordTracksEvent( `calypso_upgrade_plan_upsell_${ buttonAction }_button_click`, {
+		section: 'checkout',
+	} );
+};
+
+export default connect(
+	( state, props ) => {
+		const { selectedSiteId } = props;
+		const productsList = getProductsList( state );
+		const sitePlans = getPlansBySiteId( state ).data;
+		const planSlug = getUpgradePlanSlugFromPath( state, selectedSiteId, props.product );
+		const annualDiscountPrice = getPlanDiscountedRawPrice( state, selectedSiteId, planSlug, {
+			isMonthly: false,
+		} );
+		const annualPrice = getSitePlanRawPrice( state, selectedSiteId, planSlug, {
+			isMonthly: false,
+		} );
+		return {
+			currencyCode: getCurrentUserCurrencyCode( state ),
+			isLoading: isProductsListFetching( state ) || isRequestingSitePlans( state, selectedSiteId ),
+			hasProductsList: Object.keys( productsList ).length > 0,
+			hasSitePlans: sitePlans && sitePlans.length > 0,
+			siteSlug: getSiteSlug( state, selectedSiteId ),
+			planRawPrice: annualPrice,
```

Comment:
@niranjan-uma-shankar: I've refactored the part relating the price/discounted price from the old concierge code to this. First time doing this, and although seems to be working fine, please make sure there's nothing off here :D

Reply:
Seems fine 👍